skimage/feature/corner.py

```diff
@@ -138,7 +139,123 @@ def structure_tensor(image, sigma=1, mode='constant', cval=0, order=None):
+    #        I * dG/dx_i
+    # that is, the convolution of I with the derivative of a Gaussian.
+    # We need to call scipy.ndimage.gaussian_filter with the argument
+    # "order" which indicates the derivative order in the respective
+    # directions, where 0 = just Gaussian smoothing
+    #                   1 = convolve with first derivative of Gaussian
+    #                   etc.
+    # so supplying order=[2, 0] computes the 2nd Gaussian derivative in
+    # the first direction, and just smoothes the field in the second
+    # direction. That corresponds to the lower-right element of the
+    # Hessian matrix, Hcc. This is why below we will call the array
+    # deriv_step{1/2}[::-1] thus in reverse order, because the image
+    # will be in coordinates [(z,)y,x], but we need the Hessian in
+    # order [d^2/dx^2, d^2/(dx*dy), ...], so in reverse order.
+    for deriv_dirs in combinations_with_replacement(idx, 2):
+        # E.g., for idx=[0, 1] we get deriv_dirs=[0, 0]; [0, 1]; [1, 1]
+
+        deriv_step1 = (idx == deriv_dirs[0]).astype(1)
+        deriv_step2 = (idx == deriv_dirs[1]).astype(1)
+        # E.g., for deriv_dirs=[0, 0] we get deriv_step1=[1, 0]
+        #                                and deriv_step2=[1, 0]
+        #       for deriv_dirs=[0, 1] we get deriv_step1=[1, 0]
+        #                                and deriv_step2=[0, 1]
+        #       etc., expressing the two successive derivative
+        #             operations in the Hessian
+
+        if order == 'rc':
+            deriv_step1 = deriv_step1[::-1]
+            deriv_step2 = deriv_step2[::-1]
+            # For, e.g., deriv_order=[2, 0], we want the second
+            # derivative in the "horizontal"/"row" direction, and
+            # just Gaussian smoothing in the "vertical"/"column"
+            # direction. To do that on an array, we need to
+            # differentiate as [0, 2], because the first direction
+            # is the vertical direction, and the second the horizontal
+            # direction. Hence, we reverse the list order.
+
+        # Apply two successive Gaussian filter operations, as per detailed in
+        # https://dsp.stackexchange.com/questions/78280/are-scipy-second-order-gaussian-derivatives-correct
+        if np.isscalar(sigma):
+            sigma = (sigma,) * image.ndim
+
+        # For small values of sigma, the scipy Gaussian filter
+        # suffers from aliasing and edge artifacts, given that
+        # the filter will approximate a sinc or sinc derivative
+        # which only goes to 0 very slowly (order 1/n^2). Thus,
+        # we will use a much larger truncate value to reduce any
+        # edge artifacts.
+        truncate = 8 if all(s > 1 for s in sigma) else 100
+
+        sq1_2 = 1 / math.sqrt(2)
+        sigma_scaled = tuple(sq1_2 * s for s in sigma)
+        common_kwargs = dict(sigma=sigma_scaled, mode=mode, cval=cval,
+                             truncate=truncate)
```

Comment:
```suggestion
    if np.isscalar(sigma):
        sigma = (sigma,) * image.ndim
    # For small values of sigma, the scipy Gaussian filter
    # suffers from aliasing and edge artifacts, given that
    # the filter will approximate a sinc or sinc derivative
    # which only goes to 0 very slowly (order 1/n^2). Thus,
    # we will use a much larger truncate value to reduce any
    # edge artifacts.
    truncate = 8 if all(s > 1 for s in sigma) else 100
    sq1_2 = 1 / math.sqrt(2)
    sigma_scaled = tuple(sq1_2 * s for s in sigma)
    common_kwargs = dict(sigma=sigma_scaled, mode=mode, cval=cval,
                         truncate=truncate)

    for deriv_dirs in combinations_with_replacement(idx, 2):
        # E.g., for idx=[0, 1] we get deriv_dirs=[0, 0]; [0, 1]; [1, 1]

        deriv_step1 = (idx == deriv_dirs[0]).astype(1)
        deriv_step2 = (idx == deriv_dirs[1]).astype(1)
        # E.g., for deriv_dirs=[0, 0] we get deriv_step1=[1, 0]
        #                                and deriv_step2=[1, 0]
        #       for deriv_dirs=[0, 1] we get deriv_step1=[1, 0]
        #                                and deriv_step2=[0, 1]
        #       etc., expressing the two successive derivative
        #             operations in the Hessian

        if order == 'rc':
            deriv_step1 = deriv_step1[::-1]
            deriv_step2 = deriv_step2[::-1]
            # For, e.g., deriv_order=[2, 0], we want the second
            # derivative in the "horizontal"/"row" direction, and
            # just Gaussian smoothing in the "vertical"/"column"
            # direction. To do that on an array, we need to
            # differentiate as [0, 2], because the first direction
            # is the vertical direction, and the second the horizontal
            # direction. Hence, we reverse the list order.
            
        # Apply two successive Gaussian filter operations, as per detailed in
        # https://dsp.stackexchange.com/questions/78280/are-scipy-second-order-gaussian-derivatives-correct
```